internal/data/receivers_wallet.go

```diff
@@ -461,12 +461,13 @@ func (rw *ReceiverWalletModel) GetByStellarAccountAndMemo(ctx context.Context, s
 		FROM receiver_wallets rw
 		JOIN wallets w ON rw.wallet_id = w.id
 		WHERE rw.stellar_address = $1
+		AND w.sep_10_client_domain = $2
 	`
 
 	// append memo to query if it is not empty
```

Comment:
Related to my other comment. If ClientDomain can be empty, we should probably switch this to something like this: 

```go
         args := []interface{}{stellarAccount}
	if stellarMemo != "" {
		query += " AND rw.stellar_memo = $2"
		args = append(args, stellarMemo)
	} else {
		query += " AND (rw.stellar_memo IS NULL OR rw.stellar_memo = '')"
	}
	
	if clientDomain != "" {
		query += " AND w.sep_10_client_domain = $3"
		args = append(args, clientDomain)
	}
	```